consensus/hotstuff/voteaggregator/interfaces.go

```diff
@@ -0,0 +1,24 @@
+package voteaggregator
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// LazyInitCollector is a helper structure that is used to return collector which is lazy initialized
+type LazyInitCollector struct {
+	Collector hotstuff.VoteCollector
+	Created   bool // whether collector was created or retrieved from cache
+}
+
+type VoteCollectors interface {
+	// GetOrCreateCollector performs lazy initialization of hotstuff.VoteCollector
+	// collector is indexed by blockID and view
+	GetOrCreateCollector(view uint64, blockID flow.Identifier) (LazyInitCollector, error)
+	// PruneByView prunes already stored collectors by view.
+	PruneByView(view uint64) error
+	// ProcessBlock processes already validated block.
+	// Calling this function will mark conflicting collectors as stale and change state of valid collectors
+	ProcessBlock(block *model.Block) error
```

Comment:
I don't think Vote Collectors process blocks, they should just return VoteCollector, and letting the returned VoteCollector process the block or vote.

Reply:
`VoteCollector` doesn't process blocks, the idea that we will have _different_ vote collectors for different states, and block availability is one of the triggers where state transition can happen